chia/cmds/data_funcs.py

```diff
@@ -0,0 +1,309 @@
+from decimal import Decimal
+from pathlib import Path
+from types import TracebackType
+from typing import Dict, List, Optional, Tuple, Type
+
+import aiohttp
+
+from chia.cmds.units import units
+from chia.rpc.data_layer_rpc_client import DataLayerRpcClient
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.byte_types import hexstr_to_bytes
+from chia.util.config import load_config
+from chia.util.default_root import DEFAULT_ROOT_PATH
+from chia.util.ints import uint16, uint64
+
+# TODO: there seems to be a large amount of repetition in these to dedupe
+
+
+class get_client:
+    _port: Optional[int]
+    _client: Optional[DataLayerRpcClient] = None
+
+    def __init__(self, rpc_port: Optional[int]):
+        self._port = rpc_port
+
+    async def __aenter__(self) -> Tuple[DataLayerRpcClient, int]:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+        self_hostname = config["self_hostname"]
+        if self._port is None:
+            self._port = config["data_layer"]["rpc_port"]
+        self._client = await DataLayerRpcClient.create(self_hostname, uint16(self._port), DEFAULT_ROOT_PATH, config)
+        assert self._client is not None
+        return self._client, int(self._port)
+
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        traceback: Optional[TracebackType],
+    ) -> None:
+        if self._client is None:
+            return
+        self._client.close()
+        await self._client.await_closed()
+
+
+async def create_data_store_cmd(rpc_port: Optional[int], fee: Optional[str]) -> None:
+    final_fee = None
+    if fee is not None:
+        final_fee = uint64(int(Decimal(fee) * units["chia"]))
+    try:
+        async with get_client(rpc_port) as (client, rpc_port):
+            res = await client.create_data_store(fee=final_fee)
+            print(res)
+    except aiohttp.ClientConnectorError:
```

Comment:
Error handling code would benefit from  de-duping.

Reply:
The entire CLI system needs a big overhaul in this regard, absolutely.  Also, replace the callback-based helpers with context managers so they don't need to be aware of the code 'inside' them.